Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,260 @@
+      mem_setOf_eq, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂]
+    intro a a_1
+    rcases h with ⟨_, _⟩
+    simp_all only [exists_const]
+  · intro h
+    have haux : x ∈ (L : Set α)
+    · simp only [mem_sInter, mem_setOf_eq, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂,
+        mem_image, Subtype.exists, exists_and_right, exists_eq_right] at h
+      specialize h L hL
+      cases' h with y hy
+      use ⟨x, y⟩
+    · simp only [mem_image, mem_sInter, Subtype.exists, exists_and_right, exists_eq_right]
+      rcases haux with ⟨⟨y, hyA⟩, ⟨_, rfl⟩⟩
+      simp_all only [mem_sInter, mem_setOf_eq, forall_exists_index, and_imp,
+        forall_apply_eq_imp_iff₂, mem_image, Subtype.exists, exists_and_right, exists_eq_right,
+        exists_true_left, implies_true, forall_const, exists_const]
+
+@[simp]
+lemma coe_iInter (b : β) : (↑(⋂ (B : β), j B) : Set α) = ⋂ (B : β), (↑(j B) : Set α) := by
+  ext x
+  constructor
+  · intro a
+    simp_all only [mem_image, mem_iInter, Subtype.exists, exists_and_right, exists_eq_right]
+    intro i_1
+    cases a
+    simp_all only [exists_const]
+  · intro h
+    simp only [mem_iInter, mem_image, Subtype.exists, exists_and_right, exists_eq_right] at h
+    cases' h b with hxA hx
+    simp_all only [exists_true_left, mem_image, mem_iInter, Subtype.exists, exists_and_right,
+      exists_eq_right, implies_true, exists_const]
+
+lemma coe_contained : ↑D ⊆ A := by
+  simp only [image_subset_iff, Subtype.coe_preimage_self, subset_univ]
+
+@[simp]
+lemma coe_union_self_right_eq : A ∪ ↑D = A := by
+  simp only [union_eq_left, image_subset_iff, Subtype.coe_preimage_self, subset_univ]
+
+@[simp]
+lemma coe_union_self_left_eq : ↑D ∪ A = A := by
+  simp only [union_eq_right, image_subset_iff, Subtype.coe_preimage_self, subset_univ]
+
+@[simp]
+lemma cou_inter_self_right_eq_coe : A ∩ ↑D = ↑D := by
+  simp only [inter_eq_right, image_subset_iff, Subtype.coe_preimage_self, subset_univ]
+
+@[simp]
+lemma coe_inter_self_left_eq_coe : ↑D ∩ A = ↑D := by
+  simp only [inter_eq_left, image_subset_iff, Subtype.coe_preimage_self, subset_univ]
+
+@[simp]
+lemma coe_contained_iff : D ⊆ Subtype.val ⁻¹' ↑E ↔ D ⊆ E := by
+  constructor
+  · intro h x hx
+    simp only [image_subset_iff] at h
+    specialize h hx
+    simp only [mem_preimage, mem_image, Subtype.exists, exists_and_right, exists_eq_right,
+      Subtype.coe_eta, Subtype.coe_prop, exists_const] at h
+    exact h
+  · intro h x hx
+    simp only [mem_preimage, mem_image, Subtype.exists, exists_and_right, exists_eq_right,
+      Subtype.coe_eta, Subtype.coe_prop, exists_const]
+    exact h hx
+
+@[simp]
+lemma coe_eq_iff : (D : Set α) = ↑E ↔ D = E := by
+  simp only [subset_antisymm_iff, image_subset_iff, coe_contained_iff]
+
+lemma coe_inj (h : (↑D : Set α) = ↑E) : D = E := by
+  rw [coe_eq_iff] at h
+  exact h
+
+lemma coe_mono (h : (↑D : Set α) ⊆ ↑E) : D ⊆ E := by
```

Comment:
I think `coe_mono` would normally be for the converse implication. Maybe this is `subset_of_coe_subset_coe`?

Could you add `coe_mono` and tag it with the `@[mono]` attribute?